src/libstd/net/addr.rs

```diff
@@ -760,6 +760,14 @@ impl<'a, T: ToSocketAddrs + ?Sized> ToSocketAddrs for &'a T {
     }
 }
 
+#[stable(feature = "string_to_socket_addrs", since = "1.16.0")]
+impl ToSocketAddrs for String {
+    type Iter = vec::IntoIter<SocketAddr>;
```

Comment:
Is this line necessary ?
I don't see `Iter` type being used anywhere.
Maybe the next line should be written:
`fn to_socket_addrs(&self) -> io::Result<Iter> { `